File: 백준/Bronze/2750. 수 정렬하기/수 정렬하기(버블정렬).py
import sys
N = int(sys.stdin.readline())
arr=[]
for i in range(N):
    arr.append(int(sys.stdin.readline()))

def bubble_sort(A):
    n = len(A)
    for i in range(n-1,0,-1):
        bChanged = False
        for j in range(i):
            if(A[j]>A[j+1]):
                A[j],A[j+1]=A[j+1],A[j]
                bChanged = True
        
        if not bChanged:
            break

bubble_sort(arr)
for i in arr:
    print(i)

File: 백준/Bronze/2750. 수 정렬하기/수 정렬하기(버블정렬).js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

const N = Number(input[0]);
const arr = [];
for(let i=1; i<=N;i++){
    arr.push(Number(input[i]));
}

function bubble_sort(A){
    const n = A.length;
    for(let i = n-1; i>0; i--){
        let bChanged = false;
        for(let j = 0; j<i; j++){
            if(A[j]>A[j+1]){
                let tmp = A[j];
                A[j] = A[j+1];
                A[j+1] = tmp;
                bChanged = true;
            }
        }
        if(!bChanged){
            break;
        }
    }
}


bubble_sort(arr);

console.log(arr.join(""));
